library/ssl_tls.c

```diff
@@ -3643,6 +3643,55 @@ static int ssl_handshake_init( mbedtls_ssl_context *ssl )
     {
         ssl->handshake = mbedtls_calloc( 1, sizeof(mbedtls_ssl_handshake_params) );
     }
+#if defined(MBEDTLS_SSL_VARIABLE_BUFFER_LENGTH)
+    /* If the buffers are too small - reallocate */
+    {
```

Comment:
Consider making a new static function rather than just a new block. It should help readability if we know exactly which items from the parent block scope are used, so we know what to consider and think about when reading the code.

Maybe the function could be named something like `reallocate_if_the_buffers_are_too_small()` :p

Reply:
The problem is that both chunks use different conditions. If this was a different language, I'd use a lambda, but here - do you have an idea on how to do it in a clean, and not overcomplicated (like with pointers to functions passed) way?

Reply:
The way I'd do it is, start by passing a boolean indicating whether to use the “too small” or “too large” check, and once that works see if a simplification becomes apparent.